src/components/Clasification.jsx

```diff
@@ -0,0 +1,106 @@
+import { useSelector, useDispatch } from 'react-redux';
+import { useEffect, useState } from 'react';
+import { useNavigate } from 'react-router-dom';
+import { fetchPokemon, fetchType } from '../redux/pokemon/pokemonSlice';
+import { filterType } from './Pokemon';
+import pokemonImg from '../assets/pokemon.png';
+
+const Clasification = () => {
+  let typesFiltered;
+  const navigate = useNavigate();
+  const dispatch = useDispatch();
+  const [loadingType, setLoadingType] = useState(true);
+  const [loadingPokemon, setLoadingPokemon] = useState(true);
+  const [search, setSearch] = useState('');
+
+  const pokemon = useSelector(
+    (state) => state.pokemon.pokemon,
+  );
+
+  const types = useSelector(
+    (state) => state.pokemon.type.results,
+  );
+
+  useEffect(() => {
+    dispatch(fetchType('pokemon')).then(() => {
+      setLoadingType(false);
+    });
+  }, [dispatch]);
```

Comment:
- You have done great so far :+1: And also implementing most changes from the previous reviewers, :+1: I also saw your comment on the issue you having with dispatch. I have cloned your project locally and tried to help you figure out a solution. follow this guideline and your issue will be resolved. :+1: 

- [ ] Instead of moving your dispatch, you can conditionally call the dispatch only when the array of `pokemon` from the redux state is empty. you need to update 3 sections of your code for this to work perfectly. first, update this use effect to this. :+1: :smile: 

```sh
  useEffect(() => {
    if (pokemon.length === 0) {
      setLoadingType(true);
      dispatch(fetchType('pokemon')).then(() => {
        setLoadingType(false);
      });
    }
  }, [dispatch]);
```

Reply:
well resolved! ✔️